ros/src/waypoint_updater/waypnt_updater.py

```diff
@@ -91,69 +159,396 @@ def loop(self):
+        else:
+            # just for debug to see what we're getting
+            rospy.logdebug("same /traffic_waypoint message received.")
+        # end if else
 
-        if old_default_velocity != config['dyn_default_velocity']:
-            rospy.loginfo("waypoint_updater:dyn_vars_cb Adjusting default_"
-                          "velocity from {} to {}"
-                          .format(old_default_velocity,
-                                  config['dyn_default_velocity']))
+    def send_waypoints(self):
+        # generates the list of LOOKAHEAD_WPS waypoints based on car location
+        # for now assume waypoints form a loop
+        # waypoints loaded into a deque with deepcopy
+        # buffer of rear_buffer points to use for location search
+        # final_waypoints_start_ptr refers to waypoint_ptr which will be start
+        # of published final_waypoints list
+        # last_waypoint was the initial waypoint in previous iteration
 
-            if config['dyn_default_velocity'] > self.max_velocity:
-                rospy.logwarn("waypoint_updater:dyn_vars_cb default_velocity "
-                              "limited to max_velocity {}"
-                              .format(self.max_velocity))
-                self.default_velocity = self.max_velocity
+        start_wps_ptr = self.closest_waypoint()
+        dequelen = len(self.f_wpdeq)
+        if self.final_waypoints_start_ptr < 0:
+            # condition on startup
+            self.final_waypoints_start_ptr = start_wps_ptr
+            self.last_waypoint = self.final_waypoints_start_ptr
+            end_waypoint_in_deque = self.last_waypoint - 1
+            # load buffer of points behind into deque
+            for w_ptr in range(end_waypoint_in_deque - self.rear_buffer,
+                               end_waypoint_in_deque+1):
+                wpd = deq_waypoint(w_ptr, copy.deepcopy(self.waypoints[w_ptr]))
+                self.f_wpdeq.append(wpd)
+        else:
+            self.last_waypoint = self.final_waypoints_start_ptr
+            end_waypoint_in_deque = self.f_wpdeq[dequelen - 1].\
+                ptr_id
+            self.final_waypoints_start_ptr = start_wps_ptr
+        # end if else
+
+        points_added = 0
+        last_waypoint_needed = (self.final_waypoints_start_ptr +
+                                self.lookahead_wps - 1) % len(self.waypoints)
+        if last_waypoint_needed == end_waypoint_in_deque:
+            rospy.loginfo("no movement, not adding to deque")
+        elif last_waypoint_needed > end_waypoint_in_deque:
+            for w_ptr in range(end_waypoint_in_deque+1,
+                               last_waypoint_needed+1):
+                wpd = deq_waypoint(w_ptr, copy.deepcopy(self.waypoints[w_ptr]))
+                self.f_wpdeq.append(wpd)
+                points_added += 1
+            # end of for
+        else:
+            for w_ptr in range(end_waypoint_in_deque+1, len(self.waypoints)):
```

Comment:
This looks like loop around condition.  Can you add comments to label the main condition for each branch of the logic?